flask web frontend/app.py

```
@@ -38,11 +39,41 @@
+def inat_api_get(url, **kwargs):
+    """A rate-limited GET request helper for the iNaturalist API."""
+    global next_api_call_time
+    with api_lock:
+        now = time.time()
+        if now < next_api_call_time:
+            time.sleep(next_api_call_time - now)
+
+        try:
+            kwargs.setdefault('headers', INAT_HEADERS)
+            kwargs.setdefault('timeout', 20)
+            response = requests.get(url, **kwargs)
```

Ruff (0.14.10)

Probable use of `requests` call without timeout

(S113)